test/unit/ObjectMakrTests/processPropertyArray.js

```diff
@@ -0,0 +1,31 @@
+define(["mocks"], function (mocks) {
+    return function () {
+        var expect = require("chai").expect;
+
+        it("leaves Object types unchanged", function () {
+            // Arrange
+            var objectMaker = mocks.mockObjectMakr();
+
+            var properties = mocks.mockPropertyArray();
+
+            //Act
+            properties.Starfish = objectMaker.processPropertyArray(properties.Starfish);
+
+            // Assert
+            expect(typeof properties.Creature).to.deep.equal("object");
```

Comment:
What are you trying to test here? Right now you're checking whether `"object"` is deeply equal to `"object"`.